client/dashboard/task-list/tasks/tax.js

```diff
@@ -58,42 +61,53 @@ class Tax extends Component {
 		this.setState( this.initialState );
 	}
 
-	componentDidUpdate( prevProps ) {
+	shouldShowSuccessScreen() {
+		const { stepIndex } = this.state;
 		const {
-			generalSettings,
 			isJetpackConnected,
 			pluginsToActivate,
-			taxSettings,
+			generalSettings,
 		} = this.props;
 		const {
-			woocommerce_calc_taxes: calcTaxes,
 			woocommerce_store_address: storeAddress,
 			woocommerce_default_country: defaultCountry,
 			woocommerce_store_postcode: storePostCode,
 		} = generalSettings;
-		const { stepIndex } = this.state;
-		const currentStep = this.getSteps()[ stepIndex ];
-		const currentStepKey = currentStep && currentStep.key;
 		const isCompleteAddress = Boolean(
 			storeAddress && defaultCountry && storePostCode
 		);
-
-		// Show the success screen if all requirements are satisfied from the beginning.
-		if (
+		return (
 			stepIndex !== null &&
-			! pluginsToActivate.length &&
 			isCompleteAddress &&
+			! pluginsToActivate.length &&
 			isJetpackConnected &&
 			this.isTaxJarSupported()
-		) {
-			/* eslint-disable react/no-did-update-set-state */
-			this.setState( { stepIndex: null } );
-			/* eslint-enable react/no-did-update-set-state */
-			return;
-		}
+		);
+	}
 
-		if ( currentStepKey === 'store_location' && isCompleteAddress ) {
-			this.completeStep();
+	componentDidUpdate( prevProps ) {
+		const {
+			generalSettings,
+			isJetpackConnected,
+			taxSettings,
+			isGeneralSettingsRequesting,
+		} = this.props;
+		const { woocommerce_calc_taxes: calcTaxes } = generalSettings;
+		const { stepIndex } = this.state;
+		const currentStep = this.getSteps()[ stepIndex ];
+		const currentStepKey = currentStep && currentStep.key;
+
+		// If general settings have stopped requesting, check if we should show success screen.
+		if (
+			prevProps.isGeneralSettingsRequesting &&
+			! isGeneralSettingsRequesting
+		) {
+			if ( this.shouldShowSuccessScreen() ) {
```

Comment:
Now that you've simplified the logic a bit, could we remove this block and setting the `step` to `null` in favor of just checking `this. shouldShowSuccessScreen()` in the render method?

Reply:
Great point. I'd rather address in a follow up as I'm hesitant to fiddle with this at this stage.